spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -244,15 +244,18 @@ class Recurrent[T : ClassTag]()
      * identical elements T(output, output). One of the elements from the cell output is
      * the updated hidden. Thus the currentInput will update its hidden element with this output.
      */
-    currentInput(hidDim) = hidden
     var i = 1
+    // init state
+    currentInput(hidDim) = if (initState != null) initState
+     else hidden
     while (i <= times) {
       currentInput(inputDim) = outputCell.select(timeDim, i)
       cells(i - 1).updateOutput(currentInput)
       currentInput(hidDim) = cells(i - 1).output.toTable(hidDim)
       i += 1
     }
 
+    initState = null
```

Comment:
If we use the following code:

```
model.forward()
model.setState()
model.forward()
model.forward()
```
Then only the second forward will use initState. 
How to handle the multi layer cases?
If there are two LSTM, then the first LSTM layer should use 0 hidden state and the second LSTM layer should use the last hidden output from the previous layer?